File: 111. Minimum Depth of Binary Tree/recursion.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int min(int a, int b){
    if (a<b) return a;
    else return b;
}
int minDepth(struct TreeNode* root) {
    if (!root) return 0;
    if (!root->left && !root->right) return 1;
    if (!root->left) return minDepth(root->right)+1;
    if (!root->right) return minDepth(root->left)+1;
    return min(1+minDepth(root->left),1+minDepth(root->right));
    
}

File: 257. Binary Tree Paths/recursion.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def binaryTreePaths(self, root):
        """
        :type root: TreeNode
        :rtype: List[str]
        """
        def constructPaths(root,path):
            if root:
                path = path + str(root.val)
                if not root.right and not root.left:
                    paths.append(path)
                else:
                    path = path + '->'
                    constructPaths(root.left,path)
                    constructPaths(root.right,path)
        paths = []
        constructPaths(root,'')
        return paths
